src/useReactions.tsx

```diff
@@ -177,7 +191,53 @@ export const ReactionsProvider = ({
         return;
       }
 
-      if (event.getType() === EventType.Reaction) {
+      if (event.getType() === ElementCallReactionEventType) {
+        const content: ECallReactionEventContent = event.getContent();
+
+        const membershipEventId = content?.["m.relates_to"]?.event_id;
+        // Check to see if this reaction was made to a membership event (and the
+        // sender of the reaction matches the membership)
+        if (
+          !memberships.some(
+            (e) => e.eventId === membershipEventId && e.sender === sender,
+          )
+        ) {
+          logger.warn(
+            `Reaction target was not a membership event for ${sender}, ignoring`,
+          );
+          return;
+        }
+
+        const emoji = content.emoji?.split(/(?:)/u)?.[0];
+
+        if (!emoji) {
+          logger.warn(`Reaction had no emoji from ${reactionEventId}`);
+          return;
+        }
+
+        // One of our custom reactions
+        const reaction = {
+          ...GenericReaction,
+          emoji,
+          // If we don't find a reaction, we can fallback to the generic sound.
+          ...ReactionSet.find((r) => r.name === content.name),
+        };
+
+        setReactions((reactions) => {
+          if (reactions[sender]) {
+            // We've still got a reaction from this user, ignore it to prevent spamming
+            return reactions;
+          }
+          setTimeout(() => {
+            // Clear the reaction after some time.
+            setReactions(({ [sender]: _unused, ...remaining }) => remaining);
+          }, REACTION_ACTIVE_TIME_MS);
```

Comment:
React will log a warning if this timeout fires after the component has been unmounted

Reply:
Fixed with a set that keeps track and clears any active timeouts.